### DISCUSSION TITLE
Where to put the DLL that plugins depend on?

### DISCUSSION THREAD

**Discussion by cchangwen** [x64dbg/x64dbg#3486] [Q&A] [answered]
If put the `dll` and `dp64` in the plugins directory together, the first loading will be successful. 

However, reloading with `loadplugin` command will fail. I'm not sure if this is a bug.
```js
unloadplugin MyPlugin.dp64
loadplugin MyPlugin.dp64
```

**Comment by mrexodia** [comment]
Could you provide the plugin (or a different plugin) that reproduces the issue)?

**Reply by cchangwen** [reply]
plugin.cpp
```cpp
#include "plugin.h"
#include "lua.hpp"

bool pluginInit(PLUG_INITSTRUCT *initStruct) {
    auto L = luaL_newstate();
    lua_close(L);
    return true;
}

void pluginSetup() {
}

void pluginStop() {
}

```

@mrexodia 
A starter plugin that dynamically links to `LuaJit`. 
When I moved `lua51.dll` to the parent directory, everything went smoothly.

Can it be explained that the working directory is `plugins` when the plugin is first loaded, 
and when the command is executed, the working directory changes?

**Reply by mrexodia** [reply, marked as answer]
So there is a bug here actually, but to work around you can make a dedicated directory for your plugin:

```
plugins\LuaJit\LuaJit.dp64 (name must match folder)
plugins\LuaJit\lua51.dll
```

This way it's also cleaner because you do not need to mix files in the plugin folder.
